src/comm/XPUMathCompat.h

```diff
@@ -29,4 +29,23 @@ __MATH_FUNCTIONS_DECL__ double rsqrt(double x) {
   return sycl::rsqrt(x);
 }
 
+template <typename T>
+inline auto div(const T& a, const T& b) {
+  return a / b;
+}
+
+template <>
+inline auto div<c10::Half>(const c10::Half& a, const c10::Half& b)
+    __ubsan_ignore_float_divide_by_zero__ {
+  volatile float res = static_cast<float>(a) / static_cast<float>(b);
```

Comment:
Please add comments here for readability. Let us not make latter maintainer confused.